src/function_classes/linear.py

```diff
@@ -18,8 +18,7 @@ def _init_param_dist(self) -> D.Distribution:
         return param_dist
 
     def evaluate(self, x_batch: torch.Tensor, *params: torch.Tensor) -> torch.Tensor:
-        weights, _ = params
-        # TODO: changed this line so it would run, but it's probably wrong!
+        weights = torch.stack(params)
```

Comment:
here, `params` is of length 32, so the weights are precisely `params`, so calling `torch.stack` correctly gets the weights.

Reply:
that shouldn't be the case ... params should be a list of Tensors :thinking:, with one element that's the tensor

also later in [this file](https://github.com/in-context-learning-2024/in-context/pull/31/files#diff-c4db8ec7e68cf4494985cd75e9b70065f510030548f62a807d7c14c182725660R35) and [another](https://github.com/in-context-learning-2024/in-context/pull/31/files#diff-81c24bb6a590f0d98124d3b9536b4166094ac206f1f04d52dc57353194d2b2ddR24)

Reply:
The bad call was the following:
![image](https://github.com/in-context-learning-2024/in-context/assets/31297176/aa4839d5-28ba-46ba-9689-611732fc9358)

Expanding it to:
![image](https://github.com/in-context-learning-2024/in-context/assets/31297176/9994500e-ab84-4336-a422-05e1ba8a5b14)

seems to have fixed the issue